src/NavLink.js

```diff
@@ -0,0 +1,113 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import SafeAnchor from './SafeAnchor';
+import NavContext from './NavContext';
+import { createBootstrapComponent } from './ThemeProvider';
+
+const propTypes = {
+  /**
+   * @default 'nav-link'
+   */
+  bsPrefix: PropTypes.string,
+
+  /**
+   * The active state of the NavItem item.
+   */
+  active: PropTypes.bool,
+
+  /**
+   * The disabled state of the NavItem item.
+   */
+  disabled: PropTypes.bool,
+
+  /** The ARIA role for the `NavLink` */
+  role: PropTypes.string,
+
+  /** The HTML href attribute for the `NavLink` */
+  href: PropTypes.string,
+
+  /**
+   * Uniquely idenifies the `NavItem` amoungst its siblings,
+   * used to determine and control the active state ofthe parent `Nav`
+   */
+  eventKey: PropTypes.any,
+
+  /** @private */
+  onClick: PropTypes.func
+};
+
+const defaultProps = {
+  disabled: false
+};
+
+class NavLink extends React.Component {
+  handleClick = e => {
+    const { onClick, eventKey, href } = this.props;
+    if (onClick) onClick(e);
+    if (this.navContext) this.navContext.onSelect(String(eventKey || href));
+  };
+
+  render() {
+    return (
+      <NavContext.Consumer>
+        {navContext => {
+          const {
+            active,
+            bsPrefix,
+            disabled,
+            className,
+            eventKey,
+            href,
+            role: propsRole,
+            onClick: _,
+            ...props
+          } = this.props;
+
+          delete props.onSelect;
+
+          const navItemKey = String(eventKey || href);
```

Comment:
can we pull this out? or do we really need this? might be more consistent to just always require event keys